parse.go

```diff
@@ -4,110 +4,113 @@ import (
 	"encoding/base64"
 	"fmt"
 	"strings"
-
-	"github.com/pkg/errors"
 )
 
+// ParseCoreString parses a core string and returns a Consent object
+//
+// note: the consent string is base64 decoded.
+// Then each field is parsed and stored in a Consent object.
+// This parser is optimized for checking multiple vendors + most of the fields.
 func ParseCoreString(c string) (*Consent, error) {
 	if c == "" {
-		return nil, errors.New("string is empty")
+		return nil, fmt.Errorf("consent string is empty")
 	}
 	// extract core string
-	cs := strings.Split(c, ".")[0]
+	cs, _, _ := strings.Cut(c, ".")
 
 	var b, err = base64.RawURLEncoding.DecodeString(cs)
 	if err != nil {
-		return nil, fmt.Errorf("DecodeString failed: %s", err.Error())
+		return nil, fmt.Errorf("decode failed: %w", err)
 	}
 
 	r := NewReader(b)
 	p := &Consent{}
 	p.Version, err = r.ReadInt(6)
 	if err != nil {
-		return nil, fmt.Errorf("Version parse failed: %s", err.Error())
+		return nil, fmt.Errorf("decode failed: %w", err)
```

Comment:
Here too we can change the format of the string once and then not have all of these individual checks.

Reply:
I understand, but the normal parser is using, behind the scenes, a `ReadBits` method from another library which returns an error.
So either we need to rewrite the normal parser from scratch ( reusing the `Bits` type ), or to discard errors and return always the default value.